### PR TITLE
Add lower limit for primary and replica batch allocators timeout

### LINKED ISSUE #14941: [BUG] Validate lower limit for Allocator Timeout
### Describe the bug

With this https://github.com/opensearch-project/OpenSearch/pull/14848 we added cluster settings for allocator's timeout of which default value is `MINUS_ONE`. We would want to add a min limit to this setting such that very small values are not added to these settings.

Related PR Comments - https://github.com/opensearch-project/OpenSearch/pull/14848#discussion_r1687381948

### Related component

Other

### Expected behavior

Min limit for allocator timeout

### Additional Details

**Plugins**
Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by imRishN**
### Description
With this https://github.com/opensearch-project/OpenSearch/pull/14848 we added cluster settings for allocator's timeout of which default value is MINUS_ONE. This PR adds a min limit to this setting such that very small values are not added to these settings.

### Related Issues
Resolves #14941

### Check List
- [X] Functionality includes testing.
- ~[] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.~
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/7839)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java [hunk 1]
```diff
@@ -92,16 +92,50 @@ public class ShardsBatchGatewayAllocator implements ExistingShardsAllocator {
...
+     * Values supported is > 20 seconds or -1 to effectively disable timeout
+     */
     public static final Setting<TimeValue> PRIMARY_BATCH_ALLOCATOR_TIMEOUT_SETTING = Setting.timeSetting(
         PRIMARY_BATCH_ALLOCATOR_TIMEOUT_SETTING_KEY,
         TimeValue.MINUS_ONE,
+        TimeValue.MINUS_ONE,
+        new Setting.Validator<>() {
+            @Override
+            public void validate(TimeValue timeValue) {
+                if (timeValue.compareTo(TimeValue.timeValueSeconds(20)) < 0 && timeValue.compareTo(TimeValue.MINUS_ONE) != 0) {
```
Comment (shwetathareja) [inline server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java:106]: create a constant for TimeValue.timeValueSeconds(20)

Comment (Bukhtawar) [reply server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java:106]: Or rather 20_000 millis

Comment (imRishN) [reply server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java]: compareTo method handles the unit conversion appropriately - 
```
@Override
    public int compareTo(TimeValue timeValue) {
        double thisValue = ((double) duration) * timeUnit.toNanos(1);
        double otherValue = ((double) timeValue.duration) * timeValue.timeUnit.toNanos(1);
        return Double.compare(thisValue, otherValue);
    }
```

**Review by shwetathareja — APPROVED** [review]

**Comment by shwetathareja** [comment]
@imRishN this limit should be updated in documentation as well.

**Comment by shwetathareja** [comment]
Also add the changelog for it.

**Review by Bukhtawar — APPROVED** [review]

File: server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java (hunk 1 above)
Comment (Bukhtawar) [reply server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java]: The comment is to avoid 20 as a hard coded value

Comment (imRishN) [reply server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java]: Created a constant for it. Are you suggesting for a different min limit?

**Review by vikasvb90 — APPROVED** [review]

**Pull request merged**